zaps/0002-encrypt-push-notifications.md

```diff
@@ -68,6 +69,15 @@ bouncer (or Google/Apple's servers, for that matter).
     expired_time: datetime | None
   ```
 
+- Make sure to correctly maintain the foreign key from `PushDevice`​ to
+  `UserProfile` during export/import. If not done carefully, it can leak
+  notifications to the wrong recipients.
```

Comment:
This isn't exactly wrong but I think it overstates the danger. This table is like lots and lots of other tables in the Zulip data model in that it has a foreign key to other tables, and that if those foreign keys got corrupted in export/import then all kinds of bad things could happen, including sending data to the wrong users. I believe our export/import system is pretty good in general about getting that right, and isn't prone to that kind of bug. If that weren't true, that'd be a problem quite apart from anything this design is doing.